tests/vxlan/test_vnet_route_leak.py

```diff
@@ -0,0 +1,180 @@
+import logging
+import pytest
+import re
+
+from tests.common.utilities import wait_until
+from vnet_constants import *
+from vnet_utils import cleanup_vnet_routes, cleanup_dut_vnets, cleanup_vxlan_tunnels, \
+                       apply_dut_config_files, generate_dut_config_files
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology("t0")
+]
+
+BGP_WAIT_TIMEOUT = 120
+
+TESTING_STATUS = "Testing"
+CLEANUP_STATUS = "Cleanup"
+
+SHOW_VNET_ROUTES_CMD = "show vnet routes all"
+SHOW_BGP_SUMMARY_CMD = "show ip bgp summary"
+SHOW_BGP_ADV_ROUTES_CMD_TEMPLATE = "show ip bgp neighbor {} advertised-routes"
+RESTART_BGP_CMD = "sudo systemctl restart bgp"
+CONFIG_SAVE_CMD = "sudo config save -y"
+CONFIG_RELOAD_CMD = "sudo config reload -y"
+
+
+@pytest.fixture(scope="module")
+def configure_dut(minigraph_facts, duthost, vnet_config, vnet_test_params):
+    """
+    Setup/teardown fixture for VNET route leak test
+
+    During the setup portion, generates VNET VxLAN configurations and applies them to the DUT
+    During the teardown portion, removes all previously pushed VNET VxLAN information from the DUT
+
+    Args:
+        minigraph_facts: Minigraph information
+        duthost: DUT host object
+        vnet_config: Dictionary containing VNET configuration information
+        vnet_test_params: Dictionary containing VNET test parameters
+    """
+
+    duthost.shell("sonic-clear fdb all")
+    generate_dut_config_files(duthost, minigraph_facts, vnet_test_params, vnet_config)
+    apply_dut_config_files(duthost, vnet_test_params)
+
+    # In this case yield is used only to separate this fixture into setup and teardown portions
+    yield None
+
+    if vnet_test_params[CLEANUP_KEY]:
+        cleanup_vnet_routes(duthost, vnet_test_params)
+        cleanup_dut_vnets(duthost, minigraph_facts, vnet_config)
+        cleanup_vxlan_tunnels(duthost, vnet_test_params)
+
+        logger.info("Reloading CONFIG_DB from JSON")
+
+        duthost.shell(CONFIG_SAVE_CMD)
+        duthost.shell(RESTART_BGP_CMD)
+    else:
+        logger.info("Skipping cleanup")
+
+
+def bgp_connected(duthost):
+    """
+    Checks if BGP connections are up
+
+    BGP connections are "up" once they have received all prefixes (6400) from all neighbors
+
+    Args:
+        duthost: DUT host object
+    """
+
+    bgp_summary = duthost.shell(SHOW_BGP_SUMMARY_CMD)["stdout"].split("\n")
+    logger.debug("BGP Summary: {}".format(bgp_summary))
+
+    # Match entire line containing BGP neighbor information
+    bgp_neighbor_regex = re.compile(r"^([0-9]{1,3}\.){3}[0-9]{1,3}.*")
```

Comment:
I would put an example above this line of what the line this regex is matching should look like.

Reply:
will fix in next push